py3status/modules/scratchpad.py

```diff
@@ -0,0 +1,183 @@
+# -*- coding: utf-8 -*-
+"""
+Display number of scratchpad windows and urgency hints.
+
+Configuration parameters:
+    cache_timeout: refresh interval for i3-msg (default 5)
+    format: display format for this module
+        (default "\u232b [\?color=scratchpad {scratchpad}]")
+    thresholds: specify color thresholds to use
+        (default [(0, "darkgray"), (1, "violet")])
+
+Format placeholders:
+    {scratchpad} number of scratchpads
+    {urgent} number of urgent scratchpads
+
+Color thresholds:
+    xxx: print a color based on the value of `xxx` placeholder
+
+Optional:
+    i3ipc: an improved python library to control i3wm (async)
+    i3-msg: send messages to i3 window manager (counter)
+
+Examples:
+```
+# hide zero scratchpad
+scrathpad {
+    format = '[\?not_zero \u232b [\?color=scratchpad {scratchpad}]]'
+}
+
+# show urgent scratchpads
+scrathpad {
+    format = '\?if=urgent \u232b URGENT {urgent}'
+}
+
+# bring up scratchpads on clicks
+scratchpad {
+    on_click 1 = 'exec i3-msg scratchpad show'
+}
+```
+
+@author shadowprince (counter), cornerman (async)
+@license Eclipse Public License (counter), BSD (async)
+
+SAMPLE OUTPUT
+[{'full_text': '\u232b '}, {'full_text': u'0', 'color': '#a9a9a9'}]
+
+violet
+[{'full_text': '\u232b '}, {'full_text': u'5', 'color': '#ee82ee'}]
+
+urgent
+{'full_text': '\u232b URGENT 1', 'urgent': True}
+"""
+
+try:
+    import i3ipc
+    from threading import Thread
+
+    IPC = "i3ipc"
+except Exception:
+    IPC = "i3msg"
```

Comment:
(Self Review/Questions)
1) Check `IPC` here or in `post_config_hook` instead?
2) Import `i3ipc` + `Thread` here or in `i3ipc setup()` instead?

Reply:
`post_config_hook` I'd say

Please remember sway users here

Reply:
>`post_config_hook` I'd say

Done.

>Please remember sway users here

Untested. Should be same. This line is in `i3msg` setup. Same as `scratchpad_counter`.
```python
106 | self.tree_command = [self.parent.py3.get_wm_msg(), "-t", "get_tree"]
```

Reply:
@nwg-piotr Are you on sway? If yes, can you try this PR?

Scatchpad starter kit.
```python
order += "scratchpad auto"
order += "scratchpad async"
order += "scratchpad counter"

scratchpad auto {
    format = "\[auto {ipc}\] [\?color=scratchpad {scratchpad}]"
}

scratchpad async {
    format = "\[{ipc}\] [\?color=scratchpad {scratchpad}]"
    ipc = "i3ipc"
}

scratchpad counter {
    format = "\[{ipc}\] [\?color=scratchpad {scratchpad}]"
    ipc = "i3msg"
}
````

Reply:
It seems to work here:

![screenshot_2019-10-24_21-52-42_748069880](https://user-images.githubusercontent.com/20579136/67520297-d5000e00-f6a8-11e9-84e5-2aab6edde9d4.png)

Reply:
I added swaymsg ipc for fun. Can test swaymsg on i3. Scratchpad starter kit updated.

Sway users should see `swaymsg` instead of `i3msg` for `{ipc}` in their format too. Let me know if we don't want to add this. Cheers.